src/server/dfly_main.cc

```diff
@@ -323,13 +324,28 @@ bool RunEngine(ProactorPool* pool, AcceptServer* acceptor) {
   auto port = GetFlag(FLAGS_port);
   auto mc_port = GetFlag(FLAGS_memcache_port);
   string unix_sock = GetFlag(FLAGS_unixsocket);
+  mode_t unix_socket_perm;
+  string perm_str = GetFlag(FLAGS_unixsocketperm);
+  if (perm_str.empty()) {
+    // get umask of running process.
+    mode_t umask_val = umask(0);
+    umask(umask_val);
+    unix_socket_perm = 0777 & ~umask_val;
+  } else {
+    if (!absl::numbers_internal::safe_strtoi_base(perm_str, &unix_socket_perm, 8) ||
```

Comment:
We're not supposed to use the `numbers_internal` namespace, and it may change at any time (like version updates).
We could write some custom code I guess, or use something like:
```cpp
size_t end = 0;
std::stoi(perm_str, &end, 8);
if (end != perm_str.data() + perm_str.size()) { ... }
```

Reply:
ok